### PR TITLE
Introducing missing deployment types

### PR DESCRIPTION
**Pull request by wavesoft**
This PR introduces the type RAML type descriptions for various
deployment-related types that were missing

~:warning: **This PR Depends on #5692**~ Merged

_JIRA issues: MARATHON-7839_

### CODE REVIEW

File: docs/docs/rest-api/public/api/v2/types/deployment.raml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        description: The current (pre-deployment) state of the root goup in marathon
+      target?:
+        type: group.Group
+        description: The target (post-deployment) state of the root goup in marathon
+      steps:
+        type: array
+        items: DeploymentStep
+        description: The total number of steps to be performed in the deployment
+      version:
+        type: datetime
```
Comment (jdef) [inline docs/docs/rest-api/public/api/v2/types/deployment.raml]: maybe this should be a string (a bit more opaque vs. promising that it's going to be a datetime)?

**Comment by jdef** [comment]
happy to see this in progress. looking forward to eventually killing https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala

**Comment by wavesoft** [comment]
@jeschkies I think this PR covers the work of #5722 .

File: docs/docs/rest-api/public/api/v2/types/deploymentEventsInfo.raml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    description: A deployment action with additional status information
+    properties:
+      readinessCheckResults:
+        type: readinessCheck.TaskReadinessCheckResult[]
+        description: |
+          If there are readiness checks present for this deployment this property contains
+          their status for every task.
+
+  # Annotate the deployment plan with more information required for the event
+  DeploymentStepInfo:
```
Comment (jeschkies) [inline docs/docs/rest-api/public/api/v2/types/deploymentEventsInfo.raml]: According to `DeploymentFormats` this should include fields `id`, `version` and `steps`.

File: docs/docs/rest-api/public/api/v2/types/deploymentEventsInfo.raml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+uses:
+  strings: stringTypes.raml
+  deploymentAction: deploymentAction.raml
+  deploymentPlan: deploymentPlan.raml
+
+types:
+
+  # Annotate the deployment action with more information required for the event
+  DeploymentActionInfo:
+    type: deploymentAction.DeploymentAction
```
Comment (jeschkies) [inline docs/docs/rest-api/public/api/v2/types/deploymentEventsInfo.raml]: Why inherit from `DeploymentAction`?

File: docs/docs/rest-api/public/api/v2/types/readinessCheck.raml [hunk 4]
```diff
@@ -63,3 +63,33 @@ types:
...
+        type: integer
+        description: The HTTP status code returned (ex. 200)
+      contentType:
+        type: string
+        description: The value of the Content-Type header in the response
+      body:
+        type: string
+        description: The raw payload of body of the HTTP response
+
+  TaskReadinessCheckResult:
```
Comment (jeschkies) [inline docs/docs/rest-api/public/api/v2/types/readinessCheck.raml]: I cannot find this anywhere in the code base. Where is it used?

File: docs/docs/rest-api/public/api/v2/types/deploymentEventsInfo.raml (hunk 2 above)
Comment (wavesoft) [reply docs/docs/rest-api/public/api/v2/types/deploymentEventsInfo.raml]: They are present in the base type of `deploymentPlan.DeploymentPlan`

File: docs/docs/rest-api/public/api/v2/types/deploymentEventsInfo.raml (hunk 3 above)
Comment (wavesoft) [reply docs/docs/rest-api/public/api/v2/types/deploymentEventsInfo.raml]: The `DeploymentActionInfo` provides additional meta-information required by the event version of `DeploymentAction`. You can find the missing fields in the [`DeploymentStepInfoWrites` formatter](https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala#L188).

I Quote:

```scala
  implicit lazy val DeploymentStepInfoWrites: Writes[DeploymentStepInfo] = Writes { info =>
    def currentAction(action: DeploymentAction): JsObject = Json.obj (
      "action" -> DeploymentAction.actionName(action),
      actionInstanceOn(action.runSpec) -> action.runSpec.id,
      "readinessCheckResults" -> info.readinessChecksByApp(action.runSpec.id)
    )
    Json.obj(
      "id" -> info.plan.id,
      "version" -> info.plan.version,
      "affectedApps" -> info.plan.affectedAppIds,
      "affectedPods" -> info.plan.affectedPodIds,
      "steps" -> info.plan.steps,
      "currentActions" -> info.step.actions.map(currentAction),
      "currentStep" -> info.stepIndex,
      "totalSteps" -> info.plan.steps.size
    )
  }

```

**Comment by wavesoft** [comment]
> Thanks. Will you follow up with a removal of the handwritten formats as well?

Yes, but in a separate PR, where I will introduce all RAML formats for all the events since it will be quite lengthy.

File: docs/docs/rest-api/public/api/v2/types/deploymentEventsInfo.raml (hunk 3 above)
Comment (jeschkies) [reply docs/docs/rest-api/public/api/v2/types/deploymentEventsInfo.raml]: Hm, for me an inheritance is an `is a` relation. So it'd say `DeploymentActionInfo is a DeploymentAction`. However, I don't think this is a proper relation. I'd rather not encode this relation in the RAML.

File: docs/docs/rest-api/public/api/v2/types/deploymentEventsInfo.raml (hunk 2 above)
Comment (jeschkies) [reply docs/docs/rest-api/public/api/v2/types/deploymentEventsInfo.raml]: As I've mentioned above I would not express such a relation in the RAML.

**Pull request merged**